src/pybind/mgr/volumes/fs/operations/group.py

```diff
@@ -22,6 +22,8 @@ class Group(GroupTemplate):
     def __init__(self, fs, vol_spec, groupname):
         if groupname == Group.NO_GROUP_NAME:
             raise VolumeException(-errno.EPERM, "Operation not permitted for group '{0}' as it is an internal group.".format(groupname))
+        if groupname and groupname.lstrip() == "":
+            raise VolumeException(-errno.EINVAL, f"'{groupname}' is a bad group name")
```

Comment:
just using `isspace()` is better?

Reply:
I need to test for **empty string** (**""**) passed in from command-line as well.

Reply:
If that's the case, then I guess it should be `if not groupname....` otherwise for `empty string ( "" )` Exception won't be raised.

Reply:
I guess we can have `if  not( groupname and groupname.strip()):`

Reply:
> I need to test for **empty string** (**""**) passed in from command-line as well.

I referred to this comment. If u say we don't need to check for `empty string("")` then `isspace()` is better.

Reply:
> > I need to test for **empty string** (**""**) passed in from command-line as well.
> 
> I referred to this comment. If u say we don't need to check for `empty string("")` then `isspace()` is better.

As I said earlier, I **do** need to test for empty string as well.
And `isspace()` returns `False` for an empty string.